format/Format.py

```diff
@@ -66,12 +66,9 @@
 
 
 class Reader(object):
-
-    _format_class_ = None
-
-    def __init__(self, filenames, **kwargs):
+    def __init__(self, format_class, filenames, **kwargs):
```

Comment:
Can you keep the original constructor parameter order?  Such that filenames is the first argument?  Otherwise this is an api-level change which breaks some of our non-dxtbx code.

Reply:
@phyy-nx suspect that this is impossible without completely restructuring, as functools.partial only allows filling in from the beginning (which kinda makes sense) - 

functools.partial(func[,*args][, **keywords])

Return a new partial object which when called will behave like func called with the positional arguments args and keyword arguments keywords. If more arguments are supplied to the call, they are appended to args. If additional keyword arguments are supplied, they extend and override keywords. 

https://docs.python.org/2/library/functools.html

I note that the complete restructuring would make much greater API changes. I suspect that the changes @Anthchirp proposes are the fairly minimal change set to avoid the single global format limitation.

Reply:
In addition to what @graeme-winter says, keeping filenames as first argument would still necessitate a change in calling code, so I'm unsure how this would make it any easier?